### DISCUSSION TITLE
Transaction Formatters / EIP2718 TX Types

### DISCUSSION THREAD

**Discussion by aaronmgdr** [wevm/viem#623] [Question]
Looking through the code i see there is some support for chains having [custom transaction formatters](https://github.com/wagmi-dev/viem/blob/main/src/types/chain.ts#L7). Im wondering what the intended use of these are: Could for example the be used to extend support for other EIP2718 transaction types? 

On a related note am i correct that [transaction Serializers](https://github.com/wagmi-dev/viem/blob/main/src/utils/transaction/serializeTransaction.ts) are only used with [local Accounts](https://viem.sh/docs/clients/wallet.html#local-accounts-private-key-mnemonic-etc)?

**Comment by jxom** [comment]
That could indeed be an example! viem is designed to adapt to different Chain block/transaction/etc schemas (eg. Celo).

`serializeTransaction` can be used with any Account that can sign a transaction. A Local Account can sign a serialized transaction locally via `account.signTransaction(...)`, and a JSON-RPC Account can sign a serialized transaction via `eth_signTransaction` (not exposed yet).

**Reply by aaronmgdr** [reply]
oh ok so for for wallets a Custom Account with unique serialization abilitity can be created like https://github.com/wagmi-dev/viem/blob/main/site/docs/accounts/custom.md 

right?

**Reply by jxom** [reply]
Yep! You can also use our [`sign` utility](https://github.com/wagmi-dev/viem/blob/main/src/accounts/utils/sign.ts) in there if you are constructing the serialized transaction yourself.

**Reply by jxom** [reply]
Actually, I think we need to export `sign`. Can do that if you need it.

**Reply by aaronmgdr** [reply]
oh yeah that would be awesome

**Comment by aaronmgdr** [comment]
In creating a custom transactionSerializer ive come across a few functions / types that are not exported that would be beneficial to have. (such as the `sign` method you mentioned).  I plan to make a PR (or series of PRs? -- let me know if there is a preference). 

## functions i hope to export 

* sign
* serializeAccessList

## Types 

* SignTransactionArgs 
* SignTransactionReturnType 


## new functions extracted from existing functions

right now viem has function to go from `publicKeyToAddress` or `privateKeyToAccount` but to get from a privateKey to just an address is not possible. given there is already  `publicKeyToAddress` all that is needed is something like 
```
export function privateKeyToPublicKey(privateKey: string) {
  return toHex(secp256k1.getPublicKey(privateKey.slice(2), false))
}
```

similarly if you are open to  exposing functions to go from mneumonic to private key or hd to private key then all a custom account would need is one `customAccountFromPrivateKey` function.

**Comment by aaronmgdr** [comment]
https://github.com/wagmi-dev/viem/pull/665
